consensus/src/round_manager.rs

```diff
@@ -1037,6 +1186,40 @@ impl RoundManager {
         result
     }
 
+    // Insert ordered certificate formed by aggregating order votes
+    async fn new_ordered_cert(
+        &mut self,
+        ordered_cert: WrappedLedgerInfo,
+        quorum_cert: &QuorumCert,
+        preferred_peer: Author,
+    ) -> anyhow::Result<()> {
+        ensure!(
+            ordered_cert.commit_info().id() == quorum_cert.certified_block().id(),
+            "QuorumCert attached to order votes doesn't match"
+        );
+        if self
+            .block_store
+            .get_block(ordered_cert.commit_info().id())
+            .is_none()
+        {
+            ORDER_CERT_CREATED_WITHOUT_BLOCK_IN_BLOCK_STORE.inc();
+        }
+        self.block_store
+            .insert_quorum_cert(
```

Comment:
QC should be already inserted when receiving order votes?

Reply:
In the good path, QC should be already inserted when receiving order votes. In case there are some network failures or messages are received out of order, I wanted to tackle the corner case. So, this will make sure the quorum cert is inserted before inserted the ordered cert.